File: software_challenge_sdk/src/server_api/sc/protocol/requests/JoinPreparedRoomRequest.java
package sc.protocol.requests;

import com.thoughtworks.xstream.annotations.XStreamAlias;
import com.thoughtworks.xstream.annotations.XStreamAsAttribute;

@XStreamAlias("joinPrepared")
public class JoinPreparedRoomRequest extends JoinRoomRequest
{
	@XStreamAsAttribute
	private String	reservationCode;

        /**
         * might be needed by XStream
         */
        public JoinPreparedRoomRequest() {
        }

	public JoinPreparedRoomRequest(String reservationCode)
	{
		this.reservationCode = reservationCode;
	}

	public String getReservationCode()
	{
		return this.reservationCode;
	}
}

File: sdk/src/main/server-api/sc/protocol/requests/JoinPreparedRoomRequest.kt
package sc.protocol.requests

import com.thoughtworks.xstream.annotations.XStreamAlias
import com.thoughtworks.xstream.annotations.XStreamAsAttribute

/**
 * Join a prepared game by reservation.
 * The code is handed out by the administrative client
 * that created the game via a PrepareGameRequest.
 */
@XStreamAlias("joinPrepared")
data class JoinPreparedRoomRequest(
        @XStreamAsAttribute
        val reservationCode: String
): ILobbyRequest
